### PR TITLE
allow modifying job_metatdata in notification templates

### PR DESCRIPTION
**Pull request by MBcom**
##### SUMMARY
the change allows to use `job_metatdata` not only as json string, but also to access the properties too. So the notification templates are much more customizable.
  
The second change allows to acces the job stdout data too in notification template as part of the `job_metatdata`. This is related to #79

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - API

##### AWX VERSION
```
awx: 9.2.0
```

##### ADDITIONAL INFORMATION
where before only this was possible as notification template:
```
{{ job_friendly_name }} #{{ job.id }} had status {{ job.status }}, view details at {{ url }}

{{ job_metadata }}
```
now this is too posible:
```
{{ job_friendly_name }} #{{ job.id }} had status {{ job.status }}, view details at {{ url }}

{{ job_metadata.stdout }} #or any other property of job_metadata
```

### CODE REVIEW

File: awx/main/models/jobs.py [hunk 1]
```diff
@@ -710,13 +710,23 @@ def notification_data(self, block=5):
                                           skipped=h.skipped,
                                           rescued=h.rescued,
                                           ignored=h.ignored)
+           
+        # get joboutput for data
+        joboutput = ""
+        try:
+            # check if information is available
+            joboutput = self._result_stdout_raw(redact_sensitive=True, escape_ascii=True)
```
Comment (ryanpetrello) [inline awx/main/models/jobs.py]: Generating raw stdout in this way has the potential to be prohibitively expensive in a way that would affect AWX performance across the board.

**Review by ryanpetrello — CHANGES_REQUESTED** [review]
As mentioned by @wenottingham (https://github.com/ansible/awx/issues/79#issuecomment-374310486) we're very unlikely to update notification content to include raw job stdout.  Either use the API to fetch this data or take advantage of the external log aggregation feature to record job output in a third party system such as logstash/loggly/Splunk.

File: awx/main/models/jobs.py (hunk 1 above)
Comment (wenottingham) [reply awx/main/models/jobs.py]: It also has a very high probability of making the notification unreliable due to the amount of data being transported over arbitrary APIs.

Comment (MBcom) [reply awx/main/models/jobs.py]: Okay, I took it out for now, but I think we should find a way to at least make it available through the notifications templates  
so everyone can at least decide for himself whether or not he needs it

**Comment by ryanpetrello** [comment]
Hey @MBcom - this represents a breaking change for this API; it's not something we're likely to merge.

**Pull request closed without merging**